adding list support to dconf

wrcomb opened the pull request:
##### SUMMARY
The current module only returns a value key. I am adding the patch to support list sub-dirs as well.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
dconf

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```
- name: Read terminal profile
  dconf:
    key: "/org/gnome/terminal/legacy/profiles:/"
    state: list
  register: terminal_profiles

- debug:
    msg: "{{ terminal_profiles['values'][0] }}"

TASK [desktop : Read terminal profile] *****************************************************************************************************
ok: [localhost]

TASK [desktop : debug] *********************************************************************************************************************
ok: [localhost] => {
    "msg": ":b1dcc9dd-5262-4d8d-a863-c897e6d979b9/"
}
```

felixfontein commented on plugins/modules/system/dconf.py:
```diff
@@ -64,6 +64,7 @@
     default: present
     choices:
       - read
+      - list
```
Adding a `list` value to `states` is not allowed (some legacy code does it, and these are always bad examples). Create an `_info` module instead (`dconf_info`), or `_facts` if appropriate.

wrcomb replied:
Thank you for your response. The state name does not matter. I changed list and fixed other bot errors
```
ansible -m dconf -a 'key=/org/gnome/ state=list_sub_dirs' localhost
[WARNING]: No inventory was parsed, only implicit localhost is available
localhost | SUCCESS => {
    "changed": false,
    "values": [
        "control-center/",
        "deja-dup/",
        "desktop/",
        "eog/",
        "evolution-data-server/",
        "nautilus/",
        "settings-daemon/",
        "shell/",
        "system/",
        "terminal/"
    ]
}
```

felixfontein replied:
The name does indeed not matter. A list functionality does not belong into the module, no matter how it is called.

Please create an `_info` module. This PR will not get merged, no matter how you call that state.

Andersson007 replied:
+1 to @felixfontein 's statement
@wrcomb please close the pr and create a new one for _info module instead.
(Don't forget about unit and/or integration tests for it, all new modules must have them)

wrcomb commented:
Thank you for your response. The state name does not matter. I changed `list` and fixed other bot errors

```
ansible -m dconf -a 'key=/org/gnome/ state=list_sub_dirs' localhost
[WARNING]: No inventory was parsed, only implicit localhost is available
localhost | SUCCESS => {
    "changed": false,
    "values": [
        "control-center/",
        "deja-dup/",
        "desktop/",
        "eog/",
        "evolution-data-server/",
        "nautilus/",
        "settings-daemon/",
        "shell/",
        "system/",
        "terminal/"
    ]
}
```

acozine commented:
@wrcomb we looked at this PR during today's PR Review Day. The functionality you're proposing is useful, but it can (and should) be added in the form of a `_info` module to support a consistent user experience.

acozine commented:
See https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_best_practices.html#scoping-your-module-s
